src/server/server.rs

```diff
@@ -255,10 +254,7 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Server<T, S> {
                 try!(self.ch.raft_router.send_raft_msg(msg.take_raft()));
                 Ok(())
             }
-            MessageType::Cmd => {
-                RECV_MSG_COUNTER.with_label_values(&["cmd"]).inc();
-                self.on_raft_command(msg.take_cmd_req(), token, msg_id)
-            }
+            MessageType::Cmd => panic!("invalid message type raftCmd"),
```

Comment:
I think close the connection or return error is more appropriate. So the tikv server won't crash due to invalid message sent by client.